rust/src/delta_datafusion.rs

```diff
@@ -224,6 +257,67 @@ fn to_scalar_value(stat_val: &serde_json::Value) -> Option<datafusion::scalar::S
     }
 }
 
+fn correct_scalar_value_type(
+    value: datafusion::scalar::ScalarValue,
+    field_dt: &ArrowDataType,
+) -> datafusion::scalar::ScalarValue {
+    match field_dt {
+        ArrowDataType::Int64 => {
+            let raw_value = i64::try_from(value).unwrap();
+            ScalarValue::from(raw_value)
+        }
+        ArrowDataType::Int32 => {
+            let raw_value = i64::try_from(value).unwrap() as i32;
+            ScalarValue::from(raw_value)
+        }
+        ArrowDataType::Int16 => {
+            let raw_value = i64::try_from(value).unwrap() as i16;
+            ScalarValue::from(raw_value)
+        }
+        ArrowDataType::Int8 => {
+            let raw_value = i64::try_from(value).unwrap() as i8;
+            ScalarValue::from(raw_value)
+        }
+        ArrowDataType::Float32 => {
+            let raw_value = f64::try_from(value).unwrap() as f32;
+            ScalarValue::from(raw_value)
+        }
+        ArrowDataType::Float64 => {
+            let raw_value = f64::try_from(value).unwrap();
+            ScalarValue::from(raw_value)
+        }
+        ArrowDataType::Decimal(_, _) => {
+            let raw_value = f64::try_from(value).unwrap();
+            ScalarValue::from(raw_value)
+        }
+        ArrowDataType::Date32 => {
+            let raw_value = i64::try_from(value).unwrap() as i32;
+            ScalarValue::Date32(Some(raw_value))
+        }
+        ArrowDataType::Date64 => {
+            let raw_value = i64::try_from(value).unwrap();
+            ScalarValue::Date64(Some(raw_value))
+        }
+        ArrowDataType::Timestamp(TimeUnit::Nanosecond, None) => {
+            let raw_value = i64::try_from(value).unwrap();
+            ScalarValue::TimestampNanosecond(Some(raw_value))
+        }
+        ArrowDataType::Timestamp(TimeUnit::Microsecond, None) => {
+            let raw_value = i64::try_from(value).unwrap();
+            ScalarValue::TimestampMicrosecond(Some(raw_value))
+        }
+        ArrowDataType::Timestamp(TimeUnit::Millisecond, None) => {
+            let raw_value = i64::try_from(value).unwrap();
+            ScalarValue::TimestampMillisecond(Some(raw_value))
+        }
+        _ => unimplemented!(
```

Comment:
Better to return a `None` here and leave the statistic value out instead of panicking.

Reply:
If we are to return a None here, then I think it would be good to log an error message with `log::error!` so we don't leave unexpected performance degradation unnoticed.